src/Rate.php

```diff
@@ -38,54 +38,71 @@ class Rate extends Ups
     /**
      * @param $rateRequest
      *
-     * @throws Exception
-     *
      * @return RateResponse
+     * @throws Exception
      */
-    public function shopRates($rateRequest)
+    public function shopRates($rateRequest, string $requestOption = 'Rate'): RateResponse
```

Comment:
Could you please add a class with all requestoption strings? (old school enum)

Reply:
Ok but you can create kind of this class. It forces type-hinting instead of relying on pure string while we don't have a good PHPStan coverage.
```php
final class RateRequestOption
{
	public const RATE = 'Rate';
	public const SHOP = 'Shop';
	// ...
}
```
and use it in function definition

Reply:
Done.